### DISCUSSION TITLE
Chrome Browser Extension idea?

### DISCUSSION THREAD

**Discussion by LockMan007** [ThioJoe/YT-Spammer-Purge#1010] [Ideas]
What about a Chrome/Brave browser extension that would either hide Scammers, 
or replace their username and comment with "I AM A SCAMMER, PLEASE REPORT ME".
By replace, I just mean for you, the viewer, not actually modifying the real comment.

The best I can do for now is using Word Replacer II
The issue is, if you click to see a reply, it doesn't apply it to the message, it only applies it when the page loads once.

Here is what I started so far:

`
{
  "version": "2.0.10",
  "replacements": [
    {
      "active": true,
      "case": "Override",
      "repA": "Pinned_by_Shark_Tank_Global",
      "repB": "SCAMMER",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "⓪‬",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "⓵",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "⓺",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "⓹",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "⓪",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "±‪",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "ᴡ",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "ʜ",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "ᴀ",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "ᴛ",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "ᴘ",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "Pinned_by_",
      "repB": "SCAMMER_",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "kevinoleary6",
      "repB": "SCAMMER",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "Make $750 Per Day",
      "repB": "SCAMMER",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "Beast Promo",
      "repB": "Beast Promo (SCAM)",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "T͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "E͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "L͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "G͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "R͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "M͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "telegram",
      "repB": "SCAM",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "Pinned_by_Shark_Tank_Global",
      "repB": "SCAMMER",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "⓪‬",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "⓵",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "⓺",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "⓹",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "⓪",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "±‪",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "ᴡ",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "ʜ",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "ᴀ",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "ᴛ",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "ᴘ",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Maintain",
      "repA": "Pinned_by_",
      "repB": "SCAMMER_",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "kevinoleary6",
      "repB": "SCAMMER",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "Make $750 Per Day",
      "repB": "SCAMMER",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "Beast Promo",
      "repB": "Beast Promo (SCAM)",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "T͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "E͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "L͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "G͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "R͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "M͙",
      "repB": "-",
      "type": "Simple"
    },
    {
      "active": true,
      "case": "Override",
      "repA": "telegram",
      "repB": "SCAM",
      "type": "Simple"
    }
  ]
}
`

**Comment by xenova** [comment]
I have actually been working on something to this effect, with the goal of hiding these comments completely (instead of just replacing words). I was thinking something similar to https://github.com/ajayyy/SponsorBlock, but instead of acting on the video player, it would act on the comment section. Also, these comments could be voted on (or reported) by multiple people (like SponsorBlock) to determine whether something is spam or not (and will then be hidden for other users). 

In fact, I am currently in the process of collecting a large database of spam comments (see https://github.com/ThioJoe/YT-Spammer-Purge/discussions/1009), which could be used as a starting point to train ML models to do automatic detection (in addition to rule-based approaches).

**Comment by xenova** [comment]
Here we go: https://github.com/xenova/commentblock :)

**Reply by KendallDoesCoding** [reply]
Wow. Amazing work @xenova

**Reply by xenova** [reply]
Thanks! 😊 Hopefully it will be useful! 😄

**Reply by timothy-pitts** [reply]
Just downloaded the extension, just to be clear does the extension hide the spam comments? Block them? Report them?

**Reply by xenova** [reply]
By default, it "blurs" them (and you can click the 3 dots -> show if you want to unblur it). However, you can change the default action to be "remove" if you want them to be removed.
